Libraries/LibWeb/DOM/ParentNode.cpp

```diff
@@ -40,6 +40,33 @@ static WebIDL::ExceptionOr<Variant<GC::Ptr<Element>, GC::Ref<NodeList>>> scope_m
 
     auto selectors = maybe_selectors.value();
 
+    bool has_named_namespace = false;
+    for (auto const& selector : selectors) {
+        for (auto const& compound_selector : selector->compound_selectors()) {
+            for (auto simple_selector : compound_selector.simple_selectors) {
+                if (simple_selector.value.has<CSS::Selector::SimpleSelector::QualifiedName>() && simple_selector.qualified_name().namespace_type == CSS::Selector::SimpleSelector::QualifiedName::NamespaceType::Named) {
+                    has_named_namespace = true;
+                    break;
+                }
+
+                if (simple_selector.value.has<CSS::Selector::SimpleSelector::PseudoClassSelector>()) {
+                    for (auto const& argument_selector : simple_selector.pseudo_class().argument_selector_list) {
```

Comment:
This isn't quite right, because the selectors inside the pseudo-class could have their own pseudo-classes. That's why the other places we do similar scans all use a separate function, because it has to recurse.